ij/src/main/kotlin/org/jetbrains/academy/test/system/test/BaseIjTestClass.kt

```diff
@@ -1,8 +1,55 @@
 package org.jetbrains.academy.test.system.test
 
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiRecursiveElementWalkingVisitor
 import com.intellij.testFramework.fixtures.BasePlatformTestCase
+import org.jetbrains.kotlin.psi.KtBlockExpression
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtProperty
 
 /**
  * A Base test class to create tests with PSI without adding intellij dependency into courses directly.
  */
-open class BaseIjTestClass : BasePlatformTestCase()
+open class BaseIjTestClass : BasePlatformTestCase() {
+
+    fun findMethodsWithContent(content: String): List<String> {
+        val methodNames = mutableListOf<String>()
+        var currentMethod = ""
+        ApplicationManager.getApplication().runReadAction<Unit?> {
+            myFixture.file.accept(object : PsiRecursiveElementWalkingVisitor() {
+                override fun visitElement(element: PsiElement) {
+                    when (element) {
+                        is KtNamedFunction -> currentMethod = element.name.toString()
+                        is KtBlockExpression -> {
+                            if (element.text.contains(content)) {
+                                methodNames.add(currentMethod)
+                            }
+                        }
+                    }
+                    super.visitElement(element)
+                }
+            })
+        }
+        return methodNames
+    }
+
+    fun hasConstantWithGivenValue(elementValue: String): Boolean {
+        var existsConstant = false
+        ApplicationManager.getApplication().runReadAction<Unit?> {
+            myFixture.file.accept(object : PsiRecursiveElementWalkingVisitor() {
+                override fun visitElement(element: PsiElement) {
```

Comment:
You can reuse `extractElementsOfType` here too.

```kotlin
    private fun KtProperty.getConstValue(): String? {
        val possibleValue = extractElementsOfType(KtConstantExpression::class.java)
        if (possibleValue.isEmpty()) {
            return null
        }
        require(possibleValue.size == 1) { "Parser error! A const variable must have only one value" }
        return possibleValue.first().text.trimIndent()
    }

    fun hasConstantWithGivenValue(elementValue: String) = ApplicationManager.getApplication().runReadAction<Boolean> {
        val elements =  myFixture.file.extractElementsOfType(KtProperty::class.java)
        val a = elements.first().children
        elements.any { it.modifierList?.text?.contains("const") ?: false && it.getConstValue() == elementValue }
    }
```

Reply:
`element.text.contains(elementValue)` does not work since the variable name can contains elementValue